src/libsyntax/parse/lexer/unicode_chars.rs

```diff
@@ -0,0 +1,156 @@
+const ASCII_ARRAY: &'static [(char, &'static str)] = &[('_', "Low Line"), ('-', "Hyphen-Minus"),
+                                                       (',', "Comma"), (';', "Semicolon"),
+                                                       (':', "Colon"), ('!', "Exclamation Mark"),
+                                                       ('?', "Question Mark"),
+                                                       ('.', "Full Stop"), ('\'', "Apostrophe"),
+                                                       ('(', "Left Parenthesis"),
+                                                       (')', "Right Parenthesis"),
+                                                       ('{', "Left Curly Bracket"),
+                                                       ('}', "Right Curly Bracket"),
+                                                       ('*', "Asterisk"), ('/', "Solidus"),
+                                                       ('\\', "Reverse Solidus"),
+                                                       ('&', "Ampersand"), ('+', "Plus Sign"),
+                                                       ('<', "Less-Than Sign"),
+                                                       ('=', "Equals Sign"),
+                                                       ('>', "Greater-Than Sign"), ];
+
+const UNICODE_ARRAY: &'static [char] = &['ߺ', '﹍', '﹎', '﹏', '‐', '‑', '‒', '–', '﹘', '⁃', '˗',
```

Comment:
For readability I'd rather have an array which has the format `(unicode_char, unicode_name, ascii_char)`, and search the ascii array to find the string. Indexes are brittle and hard to tweak.